anta/tests/security.py

```diff
@@ -654,3 +655,151 @@ def test(self) -> None:
 
             if failed_log != f"{acl_name}:\n":
                 self.result.is_failure(f"{failed_log}")
+
+
+class VerifyIPSecConnHealth(AntaTest):
+    """
+    Verifies all IPv4 security connections.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all the IPv4 security connections are established in all vrf.
+    * Failure: The test will fail if IPv4 security is not configured or any of IPv4 security connections are not established in any vrf.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.security:
+      - VerifyIPSecConnHealth:
+    ```
+    """
+
+    name = "VerifyIPSecConnHealth"
+    description = "Verifies all IPv4 security connections."
+    categories: ClassVar[list[str]] = ["security"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show ip security connection vrf all")]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyIPSecConnHealth."""
+        self.result.is_success()
+        failure_conn = []
+        command_output = self.instance_commands[0].json_output["connections"]
+
+        # Check if IP security connection is configured
+        if not command_output:
+            self.result.is_failure("IPv4 security connection are not configured.")
+            return
+
+        # Iterate over all ip sec connection
+        for connection, conn_data in command_output.items():
+            state = next(iter(conn_data["pathDict"].values()))
+            if state != "Established":
+                failure_conn.append(connection)
```

Comment:
![Screenshot 2024-04-09 at 18 00 39](https://github.com/arista-netdevops-community/anta/assets/3007422/037a9595-c2f3-455e-92c1-9e016e1a791c)
 This looks a bit confusing

Reply:
maybe something like source: dest: vrf: ?